core/primitives/src/validator_signer.rs

```diff
@@ -261,7 +262,7 @@ impl EmptyValidatorSigner {
 }
 
 /// Signer that keeps secret key in memory and signs locally.
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
```

Comment:
This makes me very uneasy. The InMemoryValidator signer is supposed to stay in memory because it contains the secret key right? We wouldn't want to send it over the network to someone so it's best to leave it without serialization and deserialization.

Reply:
Yes, unfortunately it is required by `MutableConfigValue`.

Reply:
Weird, it was required by `MutableConfigValue` but now compiles without it :D

Reply:
BTW `InMemorySigner` which also contains `secret_key` already had (de)serialization.

Reply:
Removed serialization: https://github.com/near/nearcore/pull/11536/commits/7dabdeb5346bcd60cedb01cbfce2f7b72d5906a0